src/main/java/software/amazon/smithy/lsp/document/DocumentParser.java

```diff
@@ -643,33 +644,12 @@ private boolean isShapeType() {
             return false;
         }
 
-        switch (token.toString()) {
-            case "structure":
-            case "operation":
-            case "string":
-            case "integer":
-            case "list":
-            case "map":
-            case "boolean":
-            case "enum":
-            case "union":
-            case "blob":
-            case "byte":
-            case "short":
-            case "long":
-            case "float":
-            case "double":
-            case "timestamp":
-            case "intEnum":
-            case "document":
-            case "service":
-            case "resource":
-            case "bigDecimal":
-            case "bigInteger":
-                return true;
-            default:
-                return false;
-        }
+        return switch (token.toString()) {
+            case "structure", "operation", "string", "integer", "list", "map", "boolean", "enum", "union", "blob",
```

Comment:
Could we change this to a static hashset with a `contains` check?

Reply:
Maybe. I wonder how that compares to the lookup in the string pool (which I'm guessing this switch does, in some way). Or if that's relevant